operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,560 @@
+    model_file_path: String,
+}
+
+type PixelOut = f32;
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for XgboostOperator {
+    async fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>> {
+        let init_rasters = future::try_join_all(
+            self.sources
+                .rasters
+                .iter()
+                .map(|raster| raster.clone().initialize(context)),
+        )
+        .await?;
+
+        let input = init_rasters.get(0).unwrap();
+        let in_desc = input.result_descriptor();
+
+        let out_desc = RasterResultDescriptor {
+            data_type: RasterOut,
+            time: in_desc.time,
+            bbox: in_desc.bbox,
+            resolution: in_desc.resolution,
+            spatial_reference: in_desc.spatial_reference,
+
+            // where do we get this information from?
+            // test_data/dataset_defs? If so, how to load it?
+            measurement: Measurement::Classification(ClassificationMeasurement {
+                measurement: "raw".into(),
+                classes: hashmap!(0 => "Water Bodies".to_string()),
+            }),
+        };
+        let initialized_operator = InitializedXgboostOperator {
+            result_descriptor: out_desc,
+            sources: init_rasters,
+            model_file_path: self.params.model_file_path,
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+impl InitializedRasterOperator for InitializedXgboostOperator {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        // given a set of raster bands, if they have different types,
+        // where should they be converted?
+        let vec_of_rqps: Vec<Box<dyn RasterQueryProcessor<RasterType = f64>>> = self
+            .sources
+            .iter()
+            .map(|init_raster| init_raster.query_processor().unwrap().into_f64())
+            .collect();
+
+        Ok(QueryProcessorOut(Box::new(XgboostProcessor::new(
+            vec_of_rqps,
+            self.model_file_path.clone(),
+        ))))
+    }
+}
+
+struct XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    sources: Vec<Q>,
+    model_file: String,
+}
+
+impl<Q, P> XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    pub fn new(sources: Vec<Q>, model_file_path: String) -> Self {
+        Self {
+            sources,
+            model_file: std::fs::read_to_string(model_file_path).unwrap(),
+        }
+    }
+
+    async fn process_tile_async(
+        &self,
+        bands_of_tile: Vec<Result<Tile<P>, crate::error::Error>>,
+        pool: Arc<ThreadPool>,
+    ) -> Result<RasterTile2D<PixelOut>> {
+        // gather information
+        let reference_tile = bands_of_tile.first().unwrap();
```

Comment:
perform proper error handling (error variant + ?-operator)

Reply:
Changed in [ff0886b](https://github.com/geo-engine/geoengine/pull/639/commits/ff0886b6292e3fe8cf3e4addb4f5579c05cbecd4).